lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -332,12 +321,29 @@ function checkBucketPolicy(policy, requestType, canonicalID, arn, bucketOwner, l
         permission = 'allow';
     }
     let copiedStatement = JSON.parse(JSON.stringify(policy.Statement));
+
+    const ip = request ? requestUtils.getClientIp(request, config) : undefined;
+    const isSecure = request ? requestUtils.getHttpProtocolSecurity(request, config) : undefined;
+    const requestContext = request ? new RequestContext(request.headers, request.query,
+        request.bucketName, request.objectKey, ip,
+        isSecure, request.resourceType, 's3', null, null,
+        null, null, null, null, null, null, null, null, null,
+        request.objectLockRetentionDays) : undefined;
+
     while (copiedStatement.length > 0) {
         const s = copiedStatement[0];
+
         const principalMatch = _checkPrincipals(canonicalID, arn, s.Principal);
         const actionMatch = _checkBucketPolicyActions(requestType, s.Action, log);
-        const resourceMatch = _checkBucketPolicyResources(request, s.Resource, log);
-        const conditionsMatch = _checkBucketPolicyConditions(request, s.Condition, log);
+        let resourceMatch = false;
+        let conditionsMatch = false;
+        if (request) {
+            resourceMatch = _checkBucketPolicyResources(requestContext, s.Resource, log);
+            conditionsMatch = _checkBucketPolicyConditions(requestContext, s.Condition, log);
+        } else {
+            resourceMatch = true;
+            conditionsMatch = true;
+        }
```

Comment:
```suggestion
        let resourceMatch = true;
        let conditionsMatch = true;
        if (requestContext) {
            resourceMatch = _checkBucketPolicyResources(requestContext, s.Resource, log);
            conditionsMatch = _checkBucketPolicyConditions(requestContext, s.Condition, log);
        }
```
or
```suggestion
        const resourceMatch = !requestContext || _checkBucketPolicyResources(requestContext, s.Resource, log);
        const conditionsMatch = !requestContext || _checkBucketPolicyConditions(requestContext, s.Condition, log);
```
or even maybe keep the check on `requestContext` inside `_checkBucketPolicyResources` and `_checkBucketPolicyResources` (in place of the check on `request`)